crates/ethereum/evm/src/execute.rs

```diff
@@ -66,84 +69,76 @@ where
 {
     type Primitives = EthPrimitives;
 
-    type Strategy<DB: Database> = EthExecutionStrategy<DB, EvmConfig>;
-
-    fn create_strategy<DB>(&self, db: DB) -> Self::Strategy<DB>
+    fn create_strategy<'a, DB>(
+        &'a mut self,
+        db: &'a mut State<DB>,
+        block: &'a RecoveredBlock<<Self::Primitives as NodePrimitives>::Block>,
+    ) -> impl BlockExecutionStrategy<Primitives = Self::Primitives, Error = BlockExecutionError> + 'a
     where
         DB: Database,
     {
```

Comment:
cool, this should be sufficient because if we need additional logic we must add it to the trait anyway